tests/Functional/ProxyGeneratorTest.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Zenstruck\Foundry\Tests\Functional;
+
+use Zenstruck\Foundry\AnonymousFactory;
+use Zenstruck\Foundry\Tests\Fixtures\Entity\Contact;
+use Zenstruck\Foundry\Tests\Fixtures\Entity\Post;
+use Zenstruck\Foundry\Tests\Fixtures\Factories\PostFactoryWithProxyGenerator;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+final class ProxyGeneratorTest extends ProxyTest
+{
+    protected static $POST_FACTORY = PostFactoryWithProxyGenerator::class;
+
+    /**
+     * @test
+     */
+    public function can_assert_persisted(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function can_remove_and_assert_not_persisted(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function can_refetch_object_if_object_manager_has_been_cleared(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function exception_thrown_if_trying_to_refresh_deleted_object(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function can_force_set_and_save(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function can_force_set_multiple_fields(): void
+    {
+        $this->markTestSkipped('not supported');
+    }
+
+    /**
+     * @test
+     */
+    public function exception_thrown_if_trying_to_autorefresh_object_with_unsaved_changes(): void
+    {
+        $post = static::$POST_FACTORY::createOne(['title' => 'old title', 'body' => 'old body']);
+
+        $this->assertSame('old title', $post->getTitle());
+        $this->assertSame('old body', $post->getBody());
+
+        $post->setTitle('new title');
+
+        $this->expectException(\RuntimeException::class);
+
+        $post->setBody('new body');
+    }
+
+    /**
+     * @test
+     */
+    public function can_autorefresh_between_kernel_boots(): void
+    {
+        $this->markTestSkipped('not supported');
+
+        $post = static::$POST_FACTORY::createOne(['title' => 'old title', 'body' => 'old body']);
+
+        $this->assertSame('old title', $post->getTitle());
+        $this->assertSame('old body', $post->getBody());
+
+        // reboot kernel
+        static::ensureKernelShutdown();
+        static::bootKernel();
+
+        $this->assertSame('old title', $post->getTitle());
+        $this->assertSame('old body', $post->getBody());
+    }
```

Comment:
TODO: I'm not sure if it's possible to replace the value holder instance. This is required for re-fetching of entities (e.g. after the entity manager is cleared).